bindings/pyroot_experimental/PyROOT/python/ROOT/pythonization/_rdataframe.py

```diff
@@ -0,0 +1,106 @@
+# Author: Stefan Wunsch CERN  08/2018
+
+################################################################################
+# Copyright (C) 1995-2018, Rene Brun and Fons Rademakers.                      #
+# All rights reserved.                                                         #
+#                                                                              #
+# For the licensing terms see $ROOTSYS/LICENSE.                                #
+# For the list of contributors see $ROOTSYS/README/CREDITS.                    #
+################################################################################
+
+from ROOT import pythonization
+import cppyy
+
+
+def RDataFrameAsNumpy(df, columns=None, exclude=None):
+    """Read-out the RDataFrame as a collection of numpy arrays.
+
+    The values of the dataframe are read out as numpy array of the respective type
+    if the type is a fundamental type such as float or int. If the type of the column
+    is a complex type, such as your custom class or a std::array, the returned numpy
+    array contains Python objects of this type interpreted via PyROOT.
+
+    Be aware that reading out custom types is much less performant than reading out
+    fundamental types, such as int or float, which are supported directly by numpy.
+
+    The reading is performed in multiple threads if the implicit multi-threading of
+    ROOT is enabled.
+
+    Note that this is an instant action of the RDataFrame graph and will trigger the
+    event-loop.
+
+    Parameters:
+        columns: If None return all branches as columns, otherwise specify names in iterable.
+        exclude: Exclude branches from selection.
+
+    Returns:
+        dict: Dict with column names as keys and 1D numpy arrays with content as values
+    """
+    # Import numpy lazily
+    try:
+        import numpy
+    except:
+        raise ImportError("Failed to import numpy during call of RDataFrame.AsNumpy.")
+
+    # Find all column names in the dataframe if no column are specified
+    if not columns:
+        columns = [c for c in df.GetColumnNames()]
+
+    # Exclude the specified columns
+    if exclude == None:
+        exclude = []
+    columns = [col for col in columns if not col in exclude]
+
+    # Cast input node to base RNode type
+    # TODO: Can this be removed if Take works now?
+    #df_rnode = cppyy.gbl.ROOT.RDF.AsRNode(df)
+
+    # Register Take action for each column
+    result_ptrs = {}
+    for column in columns:
+        #column_type = df_rnode.GetColumnType(column)
+        column_type = df.GetColumnType(column)
+        # TODO: Does this work with only Take?
```

Comment:
Do we still need this comment here?

Reply:
Ah damn, I forgot to remove them :D Yes, it works now with `Take` and we do not need the `RNode` cast anymore.